src/plugins/intel_gpu/include/intel_gpu/primitives/data.hpp

```diff
@@ -4,37 +4,195 @@
+            return nullptr;
+        }
+
+        OPENVINO_ASSERT(mapped_weights != nullptr, "mmap object is null");
+
+        ib >> bin_offset;
+        ib >> do_precision_conversion;
+        if (do_precision_conversion) {
+            ib >> original_size;
+            ib >> make_data(&original_dtype, sizeof(ov::element::Type));
+            ib >> make_data(&curr_dtype, sizeof(ov::element::Type));
+        } else {
+            original_size = data_size;
+        }
+
+        auto mem_obj = std::make_shared<weights_mem>();
+        mem_obj->shared_buf = std::make_shared<ov::SharedBuffer<std::shared_ptr<ov::MappedMemory>>>(
+            mapped_weights->data() + bin_offset,
+            original_size,
+            mapped_weights);
+
+        if (should_run_transformations()) {
+            run_transformations(mem_obj);
+        }
+        return mem_obj;
+    }
+
+private:
+    bool do_weightless_caching = false;
+    bool do_precision_conversion = false;
+
+    size_t bin_offset = SIZE_MAX;
+    size_t original_size = SIZE_MAX;
+    ov::element::Type original_dtype = ov::element::Type_t::undefined;
+    ov::element::Type curr_dtype = ov::element::Type_t::undefined;
+
+    bool should_run_transformations() {
+        return do_precision_conversion;
+    }
+
+    void run_transformations(std::shared_ptr<weights_mem> mem_obj) {
+        size_t num_elems = original_size / original_dtype.size();
+        ov::Shape shape({num_elems});
+        auto orig_constant =
+            std::make_shared<ov::op::v0::Constant>(original_dtype, shape, mem_obj->shared_buf);
+
+        ov::ParameterVector inputParams;
+        ov::ResultVector results;
+        results.push_back(std::make_shared<ov::op::v0::Result>(orig_constant->output(0)));
+        auto model = std::make_shared<ov::Model>(results, inputParams, "aux");
+
+        ov::pass::Manager manager("Plugin:GPU:weightless_cache_transformations");
+
+        if (do_precision_conversion) {
+            precisions_map fp_convert_precision_map = {
+                {original_dtype, curr_dtype}};
+            type_to_fuse_map empty_fuse_map = {};
+            const bool keep_precision_sensitive_in_fp32 = false;
+            const bool convert_input_output_precision = false;
+            const bool store_original_precision_as_rt_attribute = true;
+            manager.register_pass<ov::pass::ConvertPrecision>(fp_convert_precision_map,
+                                                              empty_fuse_map,
+                                                              keep_precision_sensitive_in_fp32,
+                                                              convert_input_output_precision,
+                                                              store_original_precision_as_rt_attribute);
+        }
+
+        manager.run_passes(model);
+        const auto ops = model->get_ops();
+        OPENVINO_ASSERT(std::count_if(ops.begin(), ops.end(), [](const std::shared_ptr<ov::Node>& node) {
```

Comment:
Use `std::find_if` or `std::any` to stop loop at first Constant.

Reply:
Done.